source/server/listener_impl.cc

```diff
@@ -281,8 +283,7 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
           PROTOBUF_GET_MS_OR_DEFAULT(config, listener_filters_timeout, 15000)),
       continue_on_listener_filters_timeout_(config.continue_on_listener_filters_timeout()),
       listener_factory_context_(std::make_shared<PerListenerFactoryContextImpl>(
-          parent.server_, validation_visitor_, config, this, *this,
-          parent.factory_.createDrainManager(config.drain_type()))),
```

Comment:
It seems to me that we could keep this old interface by calling `parent_.server_.drainManager().createChildManager(parent_.server_.dispatcher()` here.

Why should we prefer to pass in drain_type instead?

Reply:
I think this was one of the details that changes for reasons during my development that is no longer relevant now. Will revert to maintain the original (or closer to) interface.

- [ ] Keep previous interface

Reply:
Do we still need the `createDrainManager` on the factory?

Reply:
Good call-out. We don't need this anymore and I've removed it.